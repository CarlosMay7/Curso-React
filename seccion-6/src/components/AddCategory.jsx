import PropTypes from 'prop-types';
import { useState } from "react";

export const AddCategory = ({onNewValue}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const onSubit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length <= 1) return;

        onNewValue(inputValue.trim()); 
        setInputValue('');
    }   

  return (
    <form onSubmit={onSubit} aria-label="form">
        <input type="text" placeholder="Buscar gifs" value={inputValue} onChange={onInputChange} />
    </form>
  )
}

AddCategory.propTypes = {
  onNewValue: PropTypes.func.isRequired
}