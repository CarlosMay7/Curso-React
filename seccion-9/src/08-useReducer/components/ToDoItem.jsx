export const ToDoItem = ({todo, onDeleteToDo, onToggleToDo}) => {
  return (
    <li className="list-group-item d-flex justify-content-between">
        <span 
            onDoubleClick={() => onToggleToDo(todo.id)} 
            className={`align-self-cener ${todo.done && 'text-decoration-line-through'}`}
        >
            {todo.description}
        </span>
        <button onClick={() => onDeleteToDo(todo.id)} className="btn btn-danger">Borrar</button>
    </li>
  )
}
