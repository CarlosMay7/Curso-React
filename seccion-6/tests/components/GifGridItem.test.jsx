import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components/GifGridItem"

describe('Probando <GifGridItem />', () => { 
    const title = 'Cristiano';
    const url = 'https://cristiano.jpg/';

    test('Debe hacer match con el snapshot', () => {
       const {container} =  render(<GifGridItem title={title} url={url} />);
       expect(container).toMatchSnapshot();
    })

    test('Debe mostrar la imagen con el url indicado', () => { 
        const {container} =  render(<GifGridItem title={title} url={url} />);
        screen.debug();
        expect(screen.getByRole('img').src).toBe(url);

        const {src} = screen.getByRole('img');
        expect(src).toBe(url);
    })

    test('Debe de mostar el tÃ­tulo', () => { 
        const {container} =  render(<GifGridItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();

     })
 })