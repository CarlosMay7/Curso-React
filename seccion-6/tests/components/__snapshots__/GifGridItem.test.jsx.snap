// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <GifGridItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Cristiano"
      src="https://cristiano.jpg/"
    />
    <p>
      Cristiano
    </p>
  </div>
</div>
`;
